crates/animation/src/lib.rs

```diff
@@ -239,6 +228,61 @@ pub enum AnimationActionStack {
     },
 }
 
+fn setup_controller(world: &mut World, id: EntityId, controller: &AnimationController) {
+    if !world.has_component(id, animation_binder_mask()) {
+        world
+            .add_component(id, animation_binder_mask(), vec![])
+            .unwrap();
+    }
+
+    if !world.has_component(id, animation_binder_weights()) {
+        world
+            .add_component(id, animation_binder_weights(), vec![])
+            .unwrap();
+    }
+
+    let mut stack = vec![];
```

Comment:
What's the relationship between AnimationAction and the AnimationStack?

Reply:
AnimationStack operates on AnimtionActions by index, the idea is that you could preload the animations and then blend them as you need every frame. I'm hoping this will also make it straightforward to extend the blending with procedural poses that produces a consistent result if nothing else.

I'm not sure this is the end game of the animation system but what I'm trying to be considerate about at this moment is the amount of data that needs to be transferred from guest every frame. As that's what I'm hearing is problematic with WASM in the browser. It may be that guest won't even interact with this system at this "low level" and then it might not even matter.

Reply:
Hm isn't AnimationActions and AnimationStack doing the same thing then? AnimationActions are a list of animations with parameters of how they should be blended together; AnimationStack seems to be doing the same thing? Or is there a reason why both are needed? (It's quite a while since I looked at this though)

For pre-loading we should just be able to `AnimationClipRef::XX {].peek()` to preload that animation.

Reply:
They are, the difference being one is doing stack operations with option doing a masked blend. AnimationStack makes it straightforward to produce the right order of operations. 

I tried to use AnimationActions todo the same things a few weeks ago but it felt confusing to get the right blend when transitioning.

As for the preloading, I'm thinking more like a AnimationSet rather than individual clips.